superset/common/query_context_processor.py

```diff
@@ -592,8 +592,9 @@ def join_offset_dfs(
 
             if time_grain:
                 # move the temporal column to the first column in df
-                col = df.pop(join_keys[0])
-                df.insert(0, col.name, col)
+                if join_keys:
+                    col = df.pop(join_keys[0])
+                    df.insert(0, col.name, col)
```

Comment:
Is there always going to be a single join key? Should we future-proof this by doing:

```suggestion
                for join_key in join_keys:
                    col = df.pop(join_key)
                    df.insert(0, col.name, col)
```